lib/matplotlib/path.py

```diff
@@ -128,16 +128,20 @@ def __init__(self, vertices, codes=None, _interpolation_steps=1,
         vertices = _to_unmasked_float_array(vertices)
         if vertices.ndim != 2 or vertices.shape[1] != 2:
             raise ValueError(
-                "'vertices' must be a 2D list or array with shape Nx2")
+                "'vertices' must be a 2D list or array with shape (N, 2). "
+                f"Your input has shape {vertices.shape}.")
```

Comment:
I was thinking of also having a `cbook._check_shape((None, 2), vertices=vertices)` similar to _check_isinstance, _check_in_list, _check_getitem, given that this also comes up quite often...

Reply:
Sounds reasonable.

Reply:
Is that a "please do this for this PR" or "this seems like a good idea we should think about"?

Reply:
Both ways are ok for me.

Reply:
It's a "let's try and see whether @tacaswell is going to bite the hook and do it here, but I won't block over that either".

Reply:
:fishing_pole_and_fish: